project/Dependencies.scala

```diff
@@ -25,6 +25,6 @@ object Dependencies {
     "com.vladsch.flexmark" % "flexmark-ext-yaml-front-matter" % flexmarkVersion,
   )
 
-  val newCompilerInterface = "org.scala-sbt" % "compiler-interface" % "1.4.3"
+  val newCompilerInterface = "org.scala-sbt" % "compiler-interface" % "1.7.1"
```

Comment:
Does this have implications?  🤔   I'm not sure, but it's worth considering before bumping (and might need a comment)

Reply:
At the very last, this should be "next minor version only".

Reply:
> Does this have implications?

I don't actually think so. I spoke with @adpi2 pretty in depth about this changes before making it as I was having some issues with the tests and he mentioned that if compat breaks the tests will fail that we have in place in `sbt-test`. That's actually why I do the `diagnosticCode` the way I do in `Problem` so that no new code in the newer zinc is called _unless_ something is updated and has the new zinc and can actually call this. So I think the compat tests we have in place ensure this isn't breaking for users when they are using an old sbt version.

Reply:
Yep, sounds good.  Very nice work.